internal/common/vote/vote.go

```diff
@@ -0,0 +1,191 @@
+package vote
+
+import (
+	"math/rand"
+	"time"
+)
+
+func GetVotesForRule(ruleID string, islandID int) map[int][]int {
+	votes_layout := make(map[int][]int)
+	rand.Seed(time.Now().UTC().UnixNano())
+	i := 1
+	for {
+		votes_layout[i] = []int{rand.Intn(100), rand.Intn(100)}
+		i++
+		if i > 6 {
+			break
+		}
+	}
+	return votes_layout
+}
+
+func VoteRule(ruleID string, islandID int) (int, int, bool) {
+
+	//Get votes from islands for the rule changes.
+	votes_layout := make(map[int][]int)
+	votes_layout = Get_votes_for_rule(ruleID, islandID)
+
+	//Calculate results of each island.
+	results_of_all_islands := make(map[int][]int)
+	j := 1
+	for {
+		if votes_layout[j][0] > votes_layout[j][1] {
+			results_of_all_islands[j] = []int{1, 0}
+		}
+		if votes_layout[j][0] <= votes_layout[j][1] {
+			results_of_all_islands[j] = []int{0, 1}
+		}
+		j++
+		if j > 6 {
+			break
+		}
+	}
+
+	//Announce voting result
+	num_acc := 0
+	num_rej := 0
+	var ans bool
+	for _, v := range results_of_all_islands {
+		if v[0] == 1 {
+			num_acc++
+		}
+		if v[1] == 1 {
+			num_rej++
+		}
+	}
+	if num_acc > num_rej {
+		ans = true
+	}
+	if num_acc <= num_rej {
+		ans = false
+	}
+	return num_acc, num_rej, ans
+}
+
+func Get_votes_for_elect() map[int][]int {
+	rand.Seed(time.Now().UTC().UnixNano())
+	votes_layout_e := make(map[int][]int)
+	i := 1
+	for {
+		votes_layout_e[i] = []int{rand.Intn(100), rand.Intn(100),
+			rand.Intn(100), rand.Intn(100), rand.Intn(100), rand.Intn(100)}
+		i++
+		if i > 6 {
+			break
+		}
+	}
+
+	return votes_layout_e
+}
+
+func Vote_Elect() (int, [6]int, map[int][]int) {
+
+	//Get votes from each island for the election.
+	votes_layout_e := make(map[int][]int)
+	votes_layout_e = Get_votes_for_elect()
+
+	//Calculate the preference map.
+	var order [6]int
+	var index [6]int
+	var score [6]int
+	PreferenceMap := make(map[int][]int)
+	for k, v := range votes_layout_e {
+		t := 0
+		i := 0
+		j := 0
+
+		for {
+			order[t] = v[t]
+			t++
+			if t > 5 {
+				break
+			}
+		}
+
+		for {
+
+			p := 100
+			j = 0
+
+			for {
+				if p > order[j] {
+					p = order[j]
+					index[i] = j
+				}
+
+				j++
+				if j > 5 {
+					break
+				}
+			}
+
+			j = index[i]
+			order[j] = 101
+
+			i++
+			if i > 5 {
+				break
+			}
+		}
+
+		o := 0
+		r := 0
```

Comment:
Outside of some cases (like i for a loop index, n for a count) please avoid using single letter names for variables as they are not descriptive.